homeassistant/components/izone/climate.py

```diff
@@ -144,6 +163,15 @@ def __init__(self, controller: Controller) -> None:
 
     async def async_added_to_hass(self):
         """Call on adding to hass."""
+        if self._temperature_sensor:
+            async_track_state_change(
+                self.hass, self._temperature_sensor, self._async_temp_sensor_changed
+            )
+
+            temp_sensor_state = self.hass.states.get(self._temperature_sensor)
```

Comment:
We don't allow device integrating integrations to depend on states from other integrations.

Reply:
So really what is required then is a 'template climate' device much like a 'template cover' for example. That would enable exactly this, and be permissible by the projects desirable architecture. The 'generic thermostat' unfortunately does not meet that requirement. 

Since this however works completely fine, I guess it goes in the private repository as its too broken for me to use with the internal integration....